### PR TITLE
8344906: Simplify Java version parsing in the build file

### PR DESCRIPTION
**Pull request by nlisker**
Replaces the manual versions handling with [Version](https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/lang/Runtime.Version.html).

Changes:
* Removed the methods `parseJavaVersion(String)`, `parseJdkVersion(String)` and `compareJdkVersion(String, String)`, and replaced them with `Version.parse` and `Version.compareTo` methods.
* Removed the build properties of the Java version on which the Gradle build runs because they are only used for logging. Replaced with logging them directly. Note that Gradle logs by itself the info of the runtime it uses, so there's no need to logs these manually.
* Simplified the build JDK version by working directly with `Version` instead of a `String`. This allows to store less build properties. Also used try-with-resources to close the stream. Note that the whole build JDK manual invocation hack should be replaced with the Java Toolchain.
* Simplified the `verifyJava` task, which could also be replaced with built-in Gradle tools.

/reviewers 2

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8344906](https://bugs.openjdk.org/browse/JDK-8344906): Simplify Java version parsing in the build file (**Enhancement** - P4)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1647/head:pull/1647` \
`$ git checkout pull/1647`

Update a local copy of the PR: \
`$ git checkout pull/1647` \
`$ git pull https://git.openjdk.org/jfx.git pull/1647/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1647`

View PR using the GUI difftool: \
`$ git pr show -t 1647`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1647.diff">https://git.openjdk.org/jfx/pull/1647.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1647#issuecomment-2495426974)
</details>

### CODE REVIEW

**Comment by nlisker** [comment]
@kevinrushforth @johanvos Please review or assign reviewers.

**Comment by kevinrushforth** [comment]
At a quick glance, this looks good. I'll run it through our CI and do a few local tests early next week.

Reviewers: @kevinrushforth @johanvos 

/reviewers 2

**Review by kevinrushforth — APPROVED** [review]
All of my testing looks fine, including CI and local testing.

There is one error case where we will now fail with a less informative error message than we used to, but it is a corner case where JDK 8 is used (which hasn't worked for over 7 years), so I don't see it as a problem.

In current master, running with JDK 8 will fail with the following error:

```
* What went wrong:
Execution failed for task ':verifyJava'.
> FAIL: java version mismatch: JDK version (1.8.0_371) < minimum version (22)
```

With this patch, building with JDK 8 will fail with one of two different errors, depending on whether the JDK to run gradle is also JDK or is JDK 11 or higher:

Using JDK 21 to run gradle and setting `JDK_HOME` to JDK 8:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/kcr/dev/javafx/jfx-kcr/jfx/rt/build.gradle' line: 760

* What went wrong:
A problem occurred evaluating root project 'rt'.
> Invalid version string: '1.8.0_371-ea-85-b01'
```

Using JDK 8 to run gradle and compile JavaFX:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/kcr/dev/javafx/jfx-kcr/jfx/rt/build.gradle' line: 760

* What went wrong:
A problem occurred evaluating root project 'rt'.
> java/lang/Runtime$Version
```

The latter won't be possible in gradle 9 anyway (the JDK used to run gradle 9 will need to be >= 11), so even less of a concern.

**Comment by nlisker** [comment]
Yes, `Version` doesn't parse the old 1.x JDK formats. Since the build JDK needs to be up to date anyway, I didn't see it as a problem.

**Comment by kevinrushforth** [comment]
> Since the build JDK needs to be up to date anyway, I didn't see it as a problem.

I don't see it as a problem, either.

File: build.gradle [hunk 1]
```diff
@@ -1511,12 +1477,15 @@ logger.quiet("OS_NAME: $OS_NAME")
...
-logger.quiet("jdk build number: ${jdkBuildNumber}")
+
+Version javaVersionInfo = Runtime.version()
+logger.quiet("java.runtime.version: " + javaVersionInfo)
+logger.quiet("java version: " + javaVersionInfo.feature())
+logger.quiet("java build number: " + javaVersionInfo.build().orElse(0))
+
+logger.quiet("jdk.runtime.version: " + jdkVersionInfo)
+logger.quiet("jdk version: " + jdkVersionInfo.feature())
+logger.quiet("jdk build " + javaVersionInfo.build().orElse(0))
```
Comment (arapte) [inline build.gradle:1488]: minor:  missing **:** after _jdk build_

**Review by arapte — APPROVED** [review]
LGTM, with a minor correction (shall re-approve, when changed).
It fails as expected if we use JDK older than `jfx.build.jdk.version`
```
Execution failed for task ':verifyJava'.
> FAIL: java version mismatch: JDK version (21.0.2+13-58) < minimum version (22+36)
```

logs are as expected:
```
java.runtime.version: 23+37-2369
java version: 23
java build number: 37
jdk.runtime.version: 23+37-2369
jdk version: 23
jdk build 37
```

**Review by kevinrushforth — APPROVED** [review]

**Review by arapte — APPROVED** [review]

**Pull request closed without merging**